### ISSUE TITLE
Memory error with XDG shell stable

### ISSUE THREAD

**Issue by wmww** [swaywm/sway#2421]
Sway consistently (but non-deterministically) crashes after opening a few XDG shell stable windows. It appears to be some sort of uncaught memory error, as the crashes occur in various places (two examples in attached core dumps). I'm on Arch Linux and built both sway and wlroots from the latest master. [EDIT: or did I?]

[sway-memory-issue.tar.gz](https://github.com/swaywm/sway/files/2259874/sway-memory-issue.tar.gz)

**Comment by wmww** [comment]
Update: when run in valgrind, sway runs perfectly, because of course it does.

**Comment by wmww** [comment]
This valgrind output might be relevant:
```
==7669== Invalid write of size 8
==7669==    at 0x129262: handle_xdg_shell_surface (xdg_shell.c:452)
==7669==    by 0x6C98C4D: wlr_signal_emit_safe (signal.c:29)
==7669==    by 0x6C7F777: handle_xdg_surface_commit (wlr_xdg_surface.c:344)
==7669==    by 0x6C91641: surface_commit_pending (wlr_surface.c:398)
==7669==    by 0x6C91A34: surface_commit (wlr_surface.c:470)
==7669==    by 0x978B1C7: ffi_call_unix64 (in /usr/lib/libffi.so.6.0.4)
==7669==    by 0x978AC29: ffi_call (in /usr/lib/libffi.so.6.0.4)
==7669==    by 0x6A3BA2E: ??? (in /usr/lib/libwayland-server.so.0.1.0)
==7669==    by 0x6A37F28: ??? (in /usr/lib/libwayland-server.so.0.1.0)
==7669==    by 0x6A39A41: wl_event_loop_dispatch (in /usr/lib/libwayland-server.so.0.1.0)
==7669==    by 0x6A3814B: wl_display_run (in /usr/lib/libwayland-server.so.0.1.0)
==7669==    by 0x114FF1: server_run (server.c:159)
==7669==  Address 0x12a63518 is 24 bytes after a block of size 288 in arena "client"
```

**Comment by wmww** [comment]
Ok, so it appears to a combination of accidentally leaving a wlroots AUR package installed, and failing to give the wlroots install prefix to meson, sway ended up using a different wlroots then the one I intended. When both sway and wlroots are built from the latest master, everything seems to be fine. Sorry for the trouble.

# TLDR: everything seems to be fine
